File: kotlin-back/src/main/kotlin/com/example/kotlinback/global/mail/service/GoogleEmailService.java
package com.example.kotlinback.global.mail.service;

import com.example.kotlinback.global.mail.dto.MailTo;
import lombok.RequiredArgsConstructor;
import org.springframework.mail.SimpleMailMessage;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class GoogleEmailService {
    private final JavaMailSender javaMailSender;

    public void sendEmail(MailTo mailTo) {

        SimpleMailMessage simpleMailMessage = new SimpleMailMessage();

        simpleMailMessage.setSubject(mailTo.getTitle());
        simpleMailMessage.setText(mailTo.getMessage());
        simpleMailMessage.setTo(mailTo.getAddress());
        simpleMailMessage.setFrom("exam@exam.com");

        javaMailSender.send(simpleMailMessage);
    }
}

File: kotlin-back/src/main/kotlin/com/example/kotlinback/common/mail/service/GoogleEmailService.kt
package com.example.kotlinback.common.mail.service

import com.example.kotlinback.common.mail.dto.MailTo
import org.springframework.mail.SimpleMailMessage
import org.springframework.mail.javamail.JavaMailSender
import org.springframework.mail.javamail.JavaMailSenderImpl
import org.springframework.stereotype.Service

@Service
class GoogleEmailService(
) {
    private val javaMailSender: JavaMailSender = JavaMailSenderImpl()
    fun sendEmail(mailTo: MailTo) {
        val simpleMailMessage = SimpleMailMessage()
        simpleMailMessage.subject = mailTo.title
        simpleMailMessage.text = mailTo.message
        simpleMailMessage.setTo(mailTo.address)
        simpleMailMessage.from = "exam@exam.com"
        javaMailSender.send(simpleMailMessage)
    }
}
